### ISSUE TITLE
3rd Gen. Echo Dot does not find devices, 2nd Gen. does

### ISSUE THREAD

**Issue by SVLoneStar** [bwssytems/ha-bridge#1022]
Hi - I have a 3rd Gen. Echo Dot since yesterday.
Created some new devices in HABridge today and did a discover on the 3rd Gen Dot...nothing found.
Did a "renumber" in HABridge, and tried to discover again...nothing found.
Deleted all devices from Alexa via Amazon Web Site, tried to discover again...found only my HUE bridge and the devices created there.
Re-connected my 2nd Gen Echo Dot, tried to discover - all HABridge devices found.
Should/will the 3rd Gen Dot be able to discover HABridge devices, or is this broken on Alexa's side?

Thx, Stefan

**Comment by bwssytems** [comment]
There was a posting in another thread here and their 3rd gen device worked fine. Turn on tracupnp and post the log.

**Comment by joka222** [comment]
Echo Dot 3. gen doesn‘t find any devices. It works if an older echo dot is also connected.

**Comment by SelmaUrban** [comment]
> Echo Dot 3. gen doesn‘t find any devices. It works if an older echo dot is also connected.

I can confirm that, Echo 3. Gen needs the 2nd Gen Dot. Every device is found as
'Dimmable Light' and 'Royal Philips Electronics...'
If the old dot is offline nothing works. 

Sonoff-Tasmota has the same problem with hue emulation.

**Comment by audiofreak9** [comment]
@SVLoneStar 
I remember from past discussion also that any real HUE Bridges should be off while discovering the HA-Bridge is done.

**Comment by s3rgi** [comment]
Hi, my Echo Dot 3 neither finds any device. The same thing happens to SVLoneStar: 

2018-11-24 09:00:04,371 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: SSDP M-SEARCH packet from 192.168.29.10:50000
2018-11-24 09:00:05,022 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: send upnp discovery template 1 with response address: 192.168.29.5:8081 to address: /192.168.29.10:50000
2018-11-24 09:00:05,036 [qtp18909385-87] INFO  com.bwssystems.HABridge.upnp.UpnpSettingsResource - Traceupnp: request of description.xml from: 192.168.29.10:8081 filled in with address: 192.168.29.5:8081
2018-11-24 09:00:05,673 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: send upnp discovery template 2 with response address: 192.168.29.5:8081 to address: /192.168.29.10:50000
2018-11-24 09:00:06,325 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: send upnp discovery template 3 with response address: 192.168.29.5:8081 to address: /192.168.29.10:50000
2018-11-24 09:00:06,326 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: SSDP M-SEARCH packet from 192.168.29.10:50000
2018-11-24 09:00:06,978 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: send upnp discovery template 1 with response address: 192.168.29.5:8081 to address: /192.168.29.10:50000
2018-11-24 09:00:07,629 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: send upnp discovery template 2 with response address: 192.168.29.5:8081 to address: /192.168.29.10:50000
2018-11-24 09:00:08,280 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: send upnp discovery template 3 with response address: 192.168.29.5:8081 to address: /192.168.29.10:50000
2018-11-24 09:00:08,281 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: SSDP M-SEARCH packet from 192.168.29.10:50000

**Comment by DJSynergy** [comment]
I have the same Problem here. Got my Echo Dot 3rd Gen 2 days ago.
today i installed ha-bridge, connected my harmony hub, created a button (i want to be able to switch on my lamps without any actions) and tested the function in the web interface. light goes on and off as it should, but alexa cant find any new devices.

**Comment by amoillic** [comment]
I have the exact same issue since the addition of a Gen3 Dot device - new HA-Bridge devices are not discovered by Alexa. Even with having this Gen3 device shut down, nothing works.

**Comment by Lizard00** [comment]
Same issue here. I only have one Echo Dot Gen 3 and cannot connect any HA Bridge device.

**Comment by SVLoneStar** [comment]
Hi - my 2nd Gen Echo finds all devices just well. So does the Alexa web-based discovery.
Problem is that discoveries started by the 3rd Gen Echo Dot do not find devices.

**Comment by JKLs12** [comment]
I can confirm that HA Bridge Version 5.2.1 works with Echo Dot Gen 3. BUT I had massive problems at the beginning. Alexa didn't found new devices even if HA Bridge was correctly running. Unfortunately, I can't tell what I did to get it work. I was just testing some settings and suddenly it worked (but I know that I had this exact configuration already before).
Maybe somebody that has this problem can try to change the UPNP and Web Server Port to something else and then back to the default. That is what I did before everything began to work.
Another thing I tested multiple times is to enable/disable the Link button.

**Comment by TheFatherMind** [comment]
@JKLs12  I feel like listing some of the things that you did or your current configuration and commandline might help us figure it out.

**Comment by bwssytems** [comment]
@JKLs12 Yes, please elaborate on any conditions you changed to get the gen 3 to recognize.

**Comment by TheFatherMind** [comment]
I played with @JKLs12 settings and I could not sort it out.  So instead I just did a temporary work around until someone can figure this out.  I unplugged all my Series 3's.  Plugged in and setup a series 2.  Connected that to all my devices and then put it into DND mode and muted it.  Shoved it under my couch.  Then plugged in all my Series 3's.  They now work using the Series 2 as a bridge to the ha-bridge.  (Jesus so many bridges).

**Comment by JKLs12** [comment]
@bwssytems Sure, I can do that.
It is just a simple home network with one /24 network connected to the internet via a firewall. I have one ha bridge, Echo Dot 3 and a Logitech harmony hub. The ha bridge is connected to a switch and the other two devices are connected to the same WIFI access point. The rest in the network has nothing to do with the ha bridge (just some clients, NAS etc.). Do you need to know something special about the network?

**Comment by derherbie** [comment]
My solution is to set the port in Ha-Bridge to 80, then i unable 2 settings in my fritzbox.
In Homenet Networksettings unable UPnP and the one above(doesn't know if it is really needed).
Restart the Router an enable it again follow by restart.
After the completed reboot wait 5-10 minutes than start Alexa's search. 
I tried this 3 times and everything works fine.

![screenshot_2018-12-13 fritz box 7490](https://user-images.githubusercontent.com/45845325/49929395-ae871900-fec2-11e8-9954-45066e79e555.png)

**Comment by craigski** [comment]
I don't understand why you would need to enable UPnP on router. My understanding is this allows applications to automatically configure port forwarding from internet to devices. As this discovery process is local LAN only, ie Echo devices <--> HABridge this shouldn't be required? I have UPnP disabled on my firewall for security reasons.

http://www.ciscopress.com/articles/article.asp?p=461084

Maybe you can try to disable the UPnP, reboot router and reconfirm discover works with UPnP disabled on router?

**Comment by TheFatherMind** [comment]
@craigski  he said UNenable.. And I too see no connection between the two things.  I too have already tried to set my ha-bridge to 80 and had no luck.  We are talking about the web interface right?  My understanding is that the actual bridge is on 50000.

**Comment by craigski** [comment]
@bwssytems I have got Gen3 working :)

I did some tcpdumps on Gen2 discovery, HABridge using port 8081, this works, I can see the trace.
When I tried Gen3 discovery, I noticed some requests on port 80 Pi from Gen3 devices, they were not using the port 8081 returned from HABridge for all requests.

I reconfigured HABridge to use port 80.

Looking at tcpdump trace when using port 80 for HABridge I then saw error in request

...Z...5HTTP/1.1 200 OK
Date: Fri, 14 Dec 2018 12:39:22 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Server: Jetty(9.4.z-SNAPSHOT)

54
[{"error":{"type":"101","address":"/api/","description":"**link button not pressed**"}}]

So, I tried again, using link option in HABridge, and presto, Gen3 discovers HABridge devices.

I also noticed that the discovery process is different from Gen2 vs Gen3, Gen3 seems to do multiple requests, one for each device from the HABridge. where as Gen2 seems to request all devices in one go?

So, if you are having problems with Gen3:

1) HABridge *MUST* be configured to use port 80
2) You must press the Link option before initiating a discovery from Gen3 device.

**Comment by TheFatherMind** [comment]
Okay I feel completely foolish asking this, but....

How the heck do I change the port on this.  I changed the "serverport" value in the habridge.config file.  I modified the "Web Server Port" on the GUI and the service web page ONLY comes up on port 81 (meaning it keeps reverting to 81) no matter what I do.  And yes I stopped the service before editing the config file.

Also, how does one "use the link option".  I cannot find any button referring to "link".

Thank you in advance.

**Comment by TheFatherMind** [comment]
Okay, so I had for some reason configured the port in the service file "/etc/systemd/system/ha-bridge.service" so no matter what I did it was not running on 80.  That is why it never worked for me before.  Once I overcame this error in judgement... I was able to ask my echo to find new devices and it found them with NO EFFORT.  Never had to hit any link buttons or anything like that.  Thank you @craigski you are awesome and I appreciate you.  (:

**Comment by craigski** [comment]
First check you are not running anything on port 80 already:

Also check you are not running anything already on port 80:

`sudo netstat -pln`

I had to 'sudo reboot' to get the port to change, wouldn't change from GUI. Also check you haven't started the bridge with port in command line " -Dserver.port=81 ", I'm starting mine from /etc/systemd/system/ha-bridge.service as per install guide, without that option.


Re Link button:

Go to Bridge Control >> Update Security Settings check 'Use Link Button'
Then you should have a Link button on the Bridge Devices Page, click that, then 'discover devices' from your Gen3.

Good luck!

**Comment by TheFatherMind** [comment]
In the Alexa phone app, when controlling a  dimmable light, I am now seeing an error "Device doesn't support requested value".  However, the devices appear to be working as intended.

**Comment by audiofreak9** [comment]
@TheFatherMind 
I too have been getting the error "Device doesn't support requested value" in the app.  It appears to me the app does not like a device with a zero value or no value in the "bri".  I've been working on pinpointing the root cause, but that's for another thread.

**Comment by noesberger** [comment]
> I added a new device today in the HA-Bridge and the Echo.dot is not seeing the new device. I removed all the devices in the Alexa App and tried to discover again, but it's not seeing any Device configured in the HA-Bridge. My Sonos Speakers are found without any problem.
> I also tried an upgrade to 5.2.RC2 but no success.
> 
> I can't find any device, please help me.
> All my network devices are in the same subnet

I found the problem, the firewall was active and was blocking port 1900/udp

**Comment by sanderlv** [comment]
I have the same... :-(. I run an instance on my synology in docker with port 81 on the host network. Worked very well for over a year... but then I wanted a gen3 and that does not find any devices. Going to port 80 is not an option that is reserved by my web service. Sharing port 80 is not an option due to I cant have apache listening for a docker instance. Thus I thought I would spin up a VM as a real separate VM (not docker) and install ha-bridge there on port 80. But I have tried everything... I can't find any suitable simple lightweight image which supports ha bridge (for me). Does anyone have any tips or manual how to install ha-bridge on a VM (and where to download) and/or have a prepped VM image with HA-bridge preinstalled (preferred)?

**Comment by Ulttimattum** [comment]
Im in the same boat as sanderlv is, I have a ha bridge in docker with synology and it can only listen in port 81, not the 80 cause is reserved for web service. Because we can listen in port 80 and Echo 3gen seems only working on that port, we are stuck.

Is there another way to find the devices ha bridge shows on net from alexa app without using port 80?

**Comment by sanderlv** [comment]
I ended up in running a Lubuntu VM with java 8 (! - install-uninstall-install) and ha-bridge installed on that. Works well now...

**Comment by Ulttimattum** [comment]
Ill try this afternoon the Lubuntu on a VM with synology, what a mess to only have the port 80 free for Gen3... :(
Did you need to install java in Lubuntu for HA bridge? didnt know it was needed to work it as a web service remotely.

**Comment by sanderlv** [comment]
Yes, did that as well. Tip: java is not installed by default. Do not be tempted to use the tip install command from lubuntu, but google for install for version 8 ! the follow the install from this github section.

**Comment by bjs-pdx** [comment]
I'm trying to add devices using the Alexa app on my iPad.  I have set the bridge to port 80 on the host and am using the link button but the app never finds any devices.  Has anyone found a solution that works with the app?  I don't have an echo.

**Comment by craigski** [comment]
https://github.com/bwssytems/ha-bridge/wiki/HA-Bridge-FAQs#device

**Comment by bjs-pdx** [comment]
I have an ecobee4 thermostat with Alexa...should that work?

**Comment by craigski** [comment]
My guess is no. I checked ecobee website, https://support.ecobee.com/hc/en-us/articles/360003781292-How-do-I-connect-my-Switch-to-Philips-Hue-Smart-Lights-for-use-with-Alexa-

Seem to use the Hue website, rather than local control.

**Comment by bjs-pdx** [comment]
Echo Dot 2nd gen still working?  Really don't want to buy one but seems I have no choice.

**Comment by bjs-pdx** [comment]
So if I have a dot, add the bridge, the devices will be available on all Alexa devices?

**Comment by 1iveowl** [comment]
I tried discovering the HA Bridge initiating the search from https://alexa.amazon.com. Nothing was found. I have multiple Echo devices on my local network including Echo Spot.

Then I asked a specific old Echo unit (Gen 1) to do the search. Then it worked! I had to press Link button in the HA Bridge web interface before initiating the search.

**Comment by craigski** [comment]
I have Gen 2 and Gen 3 Echo Dot, both devices currently discover habridge itself (must use link button) and the habridge devices after the habridge has been discovered.
I suggest you configure habridge to use port 80, otherwise gen 3 wont discover.
See my previous posts about this.

**Comment by sanderlv** [comment]
Gen2 worked with port 81 on a docker. Gen 3 did not. I installed ubuntu server as a vm, followed instructions to install (this git) and installed version 8 java. all runs fine then for as well gen2 as gen3.

**Comment by bjs-pdx** [comment]
Connected the gen 2 to my network and it found all my HA Bridge devices.  That worked extremely well.  However, I had to fight with the fact that the echo doesn't support complicated wifi passwords.  Apparently it only works with 0-9 and A-Z (a-z) which is a problem for my wifi since I use complex passwords.  Amazon needs to work on this.

**Comment by edelknecht** [comment]
One note. My solution because I am using Gen 1&2&3 echos.
1. Disconnect gen 3 echos from electricity. 
2. Configure und discover newly configured devices.
3. Reconnect gen 3 echos to electricity.

Done. Hassle but works fine.

**Comment by edelknecht** [comment]
Just changed my vu+ linux receiver to use port 80 for habridge and changed the Link button security settings and that works too, Thanks, graigsky & bwssytems!

**Comment by npaisnel** [comment]
OK, hands up, I admit it, 
I have not done any googling yet on this, but I have seen mention of version 5.2.1 and ver5.2 rc2 of HA BRIDGE.

I have mine running on a RaspberryPi 3b+ 

How do I check, ( what terminal command) do I issue to check my version , and how would I go about upgrading .. a link would be great.   

Was about to get a new Gen3 for my bedroom so best I look at how I upgrade/check version first.

This all started when I wanted to get my HA -Bridge system working with Apple Siri / Home.

**Comment by DrasisDev** [comment]
@bwssytems newbie here, Installed habridge on my coreelec(kodi) image running from the SD card on an android box using the guide here : http://xbmcnut.blogspot.com/2017/03/running-ha-bridge-inside-libreelec-on.html , kodi was by default on port 8080 so I just went ahead with the instructions as is (-e SERVERPORT="80"). My echo dot 3 doesn't discover my habridge devices, I dont have any other echo or philips hue devices.

**Comment by sameeraakbar** [comment]
The newer Echo devices dont support local discovery, that is HA Bridge is not useful with New echo devices. So I followed a method using Node-Red with a node-red skill. Its working beautifully. 
This video also shows the step by step method : https://www.youtube.com/watch?v=Ilj0tLcbVwA

**Comment by bhunter1** [comment]
Before switching to Node-Red, remember that would seem to defeat the purpose of using this basic light bulb emulator.
1 Using an Echo Plus with local hub and Generation 3 software, means that Alexa can interpret these basic commands without going to the cloud.  It is called Local Voice Control.  The internet is not required to interpret the speech and determine what action to take.
2 The computer running the HA-Bridge software does not have to be internet accessible.  Alexa can communicate over the local network to issue the command.
3 One doesn't have to rely on a distant web site being up and working.  I spent 4 weeks getting an approved (but unique) device working as some Chinese site was down.
I just got HA-Bridge working today (5.2.2).  I had originally configured a systemd service to use a separate userid and high level port (in the database).  I switched back to port 80 in the database and commented out the user and group in the ha-bridge.service file.  Discovery worked.  I pressed the LINK button multiple times and don't know whether that had any effect on the success.
I see mention of creating a systemd socket for port 80 as a way of getting around the root user issue.  I intend to investigate that further.
For me the HA-Bridge software worked very well.  I switched to port 80 and it worked fine.  Given the low cost of a Raspberry Pi, a dedicated device is also a very viable option to circumvent a port 80 conflict.
The HA_Bridge software is very viable with Generation 3.
